StockTrader/src/main/java/com/restResource/StockTrader/controller/SellTriggerController.java

```diff
@@ -0,0 +1,167 @@
+
+    private InvestmentRepository investmentRepository;
+
+    public SellTriggerController(
+            SellTriggerRepository sellTriggerRepository,
+            LoggingService loggingService,
+            SellTriggerService sellTriggerService,
+            InvestmentRepository investmentRepository) {
+
+        this.sellTriggerRepository = sellTriggerRepository;
+        this.loggingService = loggingService;
+        this.sellTriggerService = sellTriggerService;
+        this.investmentRepository = investmentRepository;
+    }
+
+    @PostMapping(path = "/amount")
+    public @ResponseBody
+    HttpStatus createTriggerAmount(
+            @RequestParam String userId,
+            @RequestParam String stockSymbol,
+            @RequestParam int stockAmount,
+            @RequestParam int transactionNum) {
+
+
+        loggingService.logUserCommand(
+                UserCommandLog.builder()
+                        .command(CommandType.SET_SELL_AMOUNT)
+                        .username(userId)
+                        .stockSymbol(stockSymbol)
+                        .transactionNum(transactionNum)
+                        .build());
+
+        if (stockAmount <= 0) {
+            loggingService.logErrorEvent(
+                    ErrorEventLog.builder()
+                            .command(CommandType.SET_SELL_AMOUNT)
+                            .username(userId)
+                            .stockSymbol(stockSymbol)
+                            .transactionNum(transactionNum)
+                            .errorMessage("The amount parameter must be greater than zero")
+                            .build());
+            //FIXME do we want to throw the error or send a HttpStatus message?
+            throw new IllegalArgumentException(
+                    "The amount parameter must be greater than zero.");
+        }
+
+        //we check to see if we have sufficient stock to sell. We dont care about how much we are selling it for though.
+        if (investmentRepository.removeStocks(userId, stockAmount) == 0) { //TODO logging?
+            loggingService.logErrorEvent(
+                    ErrorEventLog.builder()
+                            .command(CommandType.SET_SELL_AMOUNT)
+                            .username(userId)
+                            .stockSymbol(stockSymbol)
+                            .transactionNum(transactionNum)
+                            .errorMessage("Insufficient Stock for the Transaction - requesting to remove: " + stockAmount)
+                            .build());
+            return HttpStatus.BAD_REQUEST; //insufficient funds for the transaction.
+        }
+
+        Optional<SellTrigger> stockSellTriggerStatus = sellTriggerRepository.findByUserIdAndStockSymbol(userId, stockSymbol);
+
+        if (stockSellTriggerStatus.isPresent()) {
+            stockSellTriggerStatus.get().setStock_amount(stockAmount + stockSellTriggerStatus.get().getStock_amount());
```

Comment:
The way you're incrementing has a race condition if the stockAmount gets changed before you save. You can avoid this with an incrementStockAmount() function instead